sdk/Trace/Sampler/TraceIdRatioBasedSampler.php

```diff
@@ -45,14 +48,21 @@ public function shouldSample(
         ?API\Attributes $attributes = null,
         ?API\Links $links = null
     ): SamplingResult {
+        // TODO: Add config to adjust which spans get sampled (only default from specification is implemented)
         if (null !== $parentContext && ($parentContext->getTraceFlags() & API\SpanContext::TRACE_FLAG_SAMPLED)) {
             return new SamplingResult(SamplingResult::RECORD_AND_SAMPLED, $attributes, $links);
         }
-
-        // TODO: implement as a function of TraceID when specification is ready
-        $decision = (lcg_value() < $this->probability)
-            ? SamplingResult::RECORD_AND_SAMPLED
-            : SamplingResult::NOT_RECORD;
+        /**
+         * Since php can only store up to 63 bit positive integers
+         */
```

Comment:
I originally wanted to use 64 bits of data from the traceId, but then I realized integers must be signed in php (only 63 bits of positive integers), so I decided to reduce to using 60 bits to only have to extract 15 hex characters instead of 16.

Reply:
The only issue I can potentially think of with this is we generate a 16 hex character string on the traceID initialization:
https://github.com/open-telemetry/opentelemetry-php/blob/ce036e540f24d48f2f37994118020ff83b8e86b9/sdk/Trace/SpanContext.php#L86

This could potentially have a conflict (albeit a small one) in a large number of traces?

Reply:
I made sure to only take the last 15 Hex characters of the traceId below to be compatible with any traceId with at least 15 hex characters in case of traceId's of larger size. I am not sure how traceId's with 16 hex characters might be an issue, but I might be misunderstanding your remark.

Reply:
I think I'm thinking too much 😅  - this should be fine!

Reply:
> I decided to reduce to using 60 bits to only have to extract 15 hex characters instead of 16.

Help me understand why you can't use all 16. Would using the last character throw off the statistics of the sampling or something given the sign bit is always 0?

Reply:
Actually even the traceId limit which would become (1 << 64) -1 would become the value -1 die to the sign bit causing comparison issues. I am not sure how to use more than 63 bits of the integer without having sign issues.

Reply:
I have been trying really hacky ways to utilize the full 64-bits of the trace ID such as story the upper and lower bits separately and doing bit manipulation and carry over to do multiplication and addition, however I realized that since the probability is a float, multiplying the probably by the range limit will inevitably have less than 60-bits of significant digits as floats are 64-bit with many bits dedicated to the exponent. Therefore, changing how I manipulate the traceID will not increase the number of bits of precision I have over 60-bit due to the precision of floats.

Reply:
Do you have any other avenues that you would like to explore or are we happy with this solution now after that reasoning?

Reply:
I can't think of a better solution at the moment, and I believe the current implementation fulfills its purpose. I discussed sampling method in the sampling SIG, and there was talk about hashing the traceID before computing probability threshold in the future but for now sticking to a similar implementation to what was done in other SDKs (python in this case). Sorry, if I am responding slowly, working on a large project I got last week. I am happy with the solution as it is.

Reply:
Ok, I'm also happy with this implementation for now too.  @morrisonlevi - if you're okay with it, I think we can merge this and change later if necessary.

Reply:
I think there is a way to do it, but I won't block the merge on it 👍